### PR TITLE
fix: Update `getTransactionByBlockHashOrBlockNumAndIndex` to pass `timestamp` to MAPI calls

### LINKED ISSUE #3042: Optimize queries in `getTransactionByBlockHashAndIndex` and `getTransactionByBlockNumberAndIndex`
In the `eth_getTransactionByBlockHashAndIndex` and `eth_getTransactionByBlockNumberAndIndex` methods, currently we are fetching the block from the `blocks/{hashOrNumber}` endpoint and using the `timestamp.from` and `timestamp.to` fields of the block to construct a timestamp range for subsequent calls to `contracts/results` endpoint in mirror node. 

Instead of doing this, we can directly call `contract/results` with the block hash/number passed as a query param and remove the redundant call to `blocks/{hashOrNumber}`, in order to reduce the traffic going to the mirror node.

Examples:
- `/api/v1/contracts/results?block.number={{blockNumber}}`
- `/api/v1/contracts/results?block.hash={{blockHash}}`

### PR DESCRIPTION
**Pull request by victor-yanev**
**Description**:

- Updates the `eth_getTransactionByBlockHashAndIndex` and `eth_getTransactionByBlockNumberAndIndex` to pass a timestamp range of the block to subsequent calls to `getContractResults`

**Related issue(s)**:

Fixes #3042

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by ebadiere — COMMENTED** [review]
The "Tests" seem to fail consistently:
```
    3 failing
  
    1) HAPI Service
         should be able to reset all counter upon reinitialization of the SDK Client:
  
        AssertionError: expected 35999 to equal 36000
        + expected - actual
  
        -35999
        +36000
        
        at Context.<anonymous> (tests/lib/hapiService.spec.ts:132:48)
        at Generator.next (<anonymous>)
        at /home/runner/_work/hedera-json-rpc-relay/hedera-json-rpc-relay/packages/relay/tests/lib/hapiService.spec.ts:27:71
        at new Promise (<anonymous>)
        at __awaiter (tests/lib/hapiService.spec.ts:23:12)
        at Context.<anonymous> (tests/lib/hapiService.spec.ts:124:20)
        at processImmediate (node:internal/timers:483:21)
  
    2) Open RPC Specification
         should execute "eth_getTransactionByBlockHashAndIndex":
       Error invoking RPC: Could not find mock for: 
  {
    "method": "get",
    "url": "contracts/results?block.hash=0x3c08bbbee74d287b1dcd3f0ca6d1d2cb92c90883c4acf9747de9f3f3162ad25b999fc7e86699f60f2a3fb3ed9a646c6b&timestamp=gte:1651560386.060890949&timestamp=lte:1651560389.060890949&transaction.index=77&limit=100&order=asc"
  }
    
  
    3) Open RPC Specification
         should execute "eth_getTransactionByBlockNumberAndIndex":
       Error invoking RPC: Could not find mock for: 
  {
    "method": "get",
    "url": "contracts/results?block.number=3&timestamp=gte:1651560386.060890949&timestamp=lte:1651560389.060890949&transaction.index=77&limit=100&order=asc"
  }
    ```

**Comment by victor-yanev** [comment]
> The "Tests" seem to fail consistently:
> 
> ```
>     3 failing
>   
>     1) HAPI Service
>          should be able to reset all counter upon reinitialization of the SDK Client:
>   
>         AssertionError: expected 35999 to equal 36000
>         + expected - actual
>   
>         -35999
>         +36000
>         
>         at Context.<anonymous> (tests/lib/hapiService.spec.ts:132:48)
>         at Generator.next (<anonymous>)
>         at /home/runner/_work/hedera-json-rpc-relay/hedera-json-rpc-relay/packages/relay/tests/lib/hapiService.spec.ts:27:71
>         at new Promise (<anonymous>)
>         at __awaiter (tests/lib/hapiService.spec.ts:23:12)
>         at Context.<anonymous> (tests/lib/hapiService.spec.ts:124:20)
>         at processImmediate (node:internal/timers:483:21)
>   
>     2) Open RPC Specification
>          should execute "eth_getTransactionByBlockHashAndIndex":
>        Error invoking RPC: Could not find mock for: 
>   {
>     "method": "get",
>     "url": "contracts/results?block.hash=0x3c08bbbee74d287b1dcd3f0ca6d1d2cb92c90883c4acf9747de9f3f3162ad25b999fc7e86699f60f2a3fb3ed9a646c6b&timestamp=gte:1651560386.060890949&timestamp=lte:1651560389.060890949&transaction.index=77&limit=100&order=asc"
>   }
>     
>   
>     3) Open RPC Specification
>          should execute "eth_getTransactionByBlockNumberAndIndex":
>        Error invoking RPC: Could not find mock for: 
>   {
>     "method": "get",
>     "url": "contracts/results?block.number=3&timestamp=gte:1651560386.060890949&timestamp=lte:1651560389.060890949&transaction.index=77&limit=100&order=asc"
>   }
>     ```
> ```

Fixed

File: packages/relay/tests/lib/eth/eth-helpers.ts [hunk 1]
```diff
@@ -25,15 +25,52 @@ import constants from '../../../src/lib/constants';
...
 import HAPIService from '../../../src/lib/services/hapiService/hapiService';
 import MockAdapter from 'axios-mock-adapter';
-import { MirrorNodeClient } from '../../../src/lib/clients/mirrorNodeClient';
+import { MirrorNodeClient } from '../../../src/lib/clients';
 import { EthImpl } from '../../../src/lib/eth';
+import EventEmitter from 'events';
 
-export function contractResultsByNumberByIndexURL(number: number, index: number): string {
-  return `contracts/results?block.number=${number}&transaction.index=${index}&limit=100&order=asc`;
+export function getQueryParams(params: object) {
```
Comment (Nana-EC) [inline packages/relay/tests/lib/eth/eth-helpers.ts:32]: Doesn't this function already exist in `helpers.ts`?
Could it be reused?

File: packages/relay/tests/lib/eth/eth-helpers.ts [hunk 2]
```diff
@@ -62,8 +99,9 @@ export function generateEthTestEnv(fixedFeeHistory = false) {
   const duration = constants.HBAR_RATE_LIMIT_DURATION;
   const total = constants.HBAR_RATE_LIMIT_TINYBAR;
   const hbarLimiter = new HbarLimit(logger.child({ name: 'hbar-rate-limit' }), Date.now(), total, duration, registry);
+  const eventEmitter = new EventEmitter();
 
-  const hapiServiceInstance = new HAPIService(logger, registry, hbarLimiter, cacheService);
+  const hapiServiceInstance = new HAPIService(logger, registry, hbarLimiter, cacheService, eventEmitter);
```
Comment (Nana-EC) [inline packages/relay/tests/lib/eth/eth-helpers.ts:104]: Seems like an unrelated change, is this a bug you realized that you're fixing.
Is so please update the PR description with the details.

**Review by ebadiere — APPROVED** [review]

File: packages/relay/tests/lib/eth/eth-helpers.ts (hunk 1 above)
Comment (victor-yanev) [reply packages/relay/tests/lib/eth/eth-helpers.ts:32]: Yes, it could be reused but it needs to be extended to also support arrays for values in the `params` object, I will extend it and remove this one. Thanks!

File: packages/relay/tests/lib/eth/eth-helpers.ts (hunk 2 above)
Comment (victor-yanev) [reply packages/relay/tests/lib/eth/eth-helpers.ts:104]: Yes, `eth-helpers.ts` wasn't updated to pass the new `EventEmitter` parameter of `HAPIService` when those changes which modified the constructor were introduced, and I was getting TypeScript errors for this file.

**Comment by victor-yanev** [comment]
After discussing this with xin, it seems like those changes were not needed
